### PR TITLE
Implement BP5 Variable Chunking Tool for Large Dataset Processing

### PR DESCRIPTION
**Pull request by sohamvsonar**
## Overview
 -  This PR introduces the read_variable_chunk tool to address critical memory limitations when processing large scientific datasets from ADIOS BP5 files. The  implementation enables AI-driven analysis of variables with millions of elements that were previously impossible to process due to memory constraints and AI  token limits.

##  Key Features:
  - Memory-efficient chunking: Automatically chunks variables >1000 elements along the first dimension
  - Dynamic sizing: Calculates optimal chunk sizes (100-500 elements) based on variable characteristics
  - Progressive reading: Provides iteration metadata for seamless chunk navigation
  - AI-friendly output: Rich metadata enables intelligent processing decisions
  - ADIOS2 integration: Uses selection API for efficient I/O operations

###  Core Implementation:
  - mcps/Adios/src/implementation/bp5_read_variable_chunk.py: New chunking implementation with automatic sizing
  - mcps/Adios/src/mcp_handlers.py: Added async wrapper with metadata handling
  - mcps/Adios/src/server.py: Tool registration and comprehensive API documentation

###  Architecture & Documentation:
  - mcps/Adios/docs/bp5_variable_chunking.md: Complete technical specification and usage guide
  - Updated tool descriptions and parameter validation across handlers

###  Testing:
  - mcps/Adios/tests/test_bp5_read_variable_chunk.py: Comprehensive test suite covering chunking logic, edge cases, and error handling
  - Enhanced existing test coverage for integration scenarios

 ### Project Configuration:
  - mcps/Adios/pyproject.toml:  Fixed Python version requirements for consistency

  Resolves: #254  - Add BP5 Variable Chunking Support: Adios MCP

### CODE REVIEW

File: mcps/Adios/docs/bp5_variable_chunking.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+- `src/implementation/bp5_read_variable_chunk.py`: Core chunking implementation
+- `src/mcp_handlers.py:108-148`: MCP handler integration  
+- `src/server.py:109-118`: Tool registration and API
+
+### Chunking Algorithm
+
+The chunking system works along the first dimension of variables:
+
+1. **Chunk Size Calculation**: `max(100, min(500, total_elements // 100))`
```
[the remark this answers is not in the document]
Comment (sohamvsonar) [reply mcps/Adios/docs/bp5_variable_chunking.md:26]: Yes, the documentation was not updated after changing it to 1%, will push the updated doc.

File: mcps/Adios/docs/bp5_variable_chunking.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+# BP5 Variable Chunking
+
+## Overview
+
+The BP5 chunking capability provides memory-efficient reading of large variables from ADIOS BP5 files. When variables contain more than 1000 elements, the system automatically switches to chunked reading to prevent memory overflows and token limit issues in AI interactions.
+
+## When Chunking is Used
+
+Chunking is automatically triggered when:
```
Comment (mattjala) [inline mcps/Adios/docs/bp5_variable_chunking.md:9]: Where is chunking automatically triggered? It looks like it's up to the MCP client to use `inpsect_variables` to manually decide when to use the chunked tools based on variable size.

File: mcps/Adios/pyproject.toml [hunk 3]
```diff
@@ -3,7 +3,7 @@ name = "adios-mcp"
 version = "1.0.0"
 description = "Fetch and analyze BP5 data files using ADIOS2. Access scientific data, metadata, and attributes for research and analysis purposes."
 readme = "README.md"
-requires-python = ">=3.10"
```
Comment (mattjala) [inline mcps/Adios/pyproject.toml:6]: What's the motivation for this? It's not necessarily a bad idea, but this is different from all the other MCPs and could lead to annoyances when switching between them.
